code/game/objects/items/weapons/RSF.dm

```diff
@@ -179,4 +183,7 @@ RSF
 		R.cell.charge -= 100
 	else
 		matter--
-		desc = "A self recharging cookie fabricator. It currently holds [matter]/10 cookie-units."
\ No newline at end of file
+		desc = "A self recharging cookie fabricator. It currently holds [matter]/10 cookie-units."
+	cooldown = 1
+	spawn(cooldowndelay)
```

Comment:
use cooldown = worldtime+cooldowndelay

and if (cooldown > worldtime)

instead of this spawn